From: Jayamohan Kallickal <jayamohan.kallickal@emulex.com>
Date: Tue, 15 Nov 2011 12:51:34 +0530
Subject: be2iscsi: memset wrb for ring create
Patch-Mainline: 3.2
References: bnc#732178
Git-commit: 37609766bf0f84b65b2181c17a7490eef139cdd1
 
This patch fixes a situation when  wrb was not being memset to zero
before being used
 
Signed-off-by: Jayamohan Kallickal <jayamohan.kallickal@emulex.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/be2iscsi/be_cmds.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/scsi/be2iscsi/be_cmds.c b/drivers/scsi/be2iscsi/be_cmds.c
index b8a82f2..2e49bba 100644
--- a/drivers/scsi/be2iscsi/be_cmds.c
+++ b/drivers/scsi/be2iscsi/be_cmds.c
@@ -660,6 +660,7 @@ int beiscsi_cmd_mccq_create(struct beiscsi_hba *phba,
 	spin_lock(&phba->ctrl.mbox_lock);
 	ctrl = &phba->ctrl;
 	wrb = wrb_from_mbox(&ctrl->mbox_mem);
+	memset(wrb, 0, sizeof(*wrb));
 	req = embedded_payload(wrb);
 	ctxt = &req->context;
 
-- 
1.6.0.2

